PostPairsTrading.md

```diff
@@ -0,0 +1,323 @@
+
+### ADF testing
+
+Imagine **we selected 13 world indexes** and aimed to assess whether they are **cointegrated or not**. In this scenario, a crucial tool at our disposal is the Augmented Dickey-Fuller (ADF) test.
+
+> 💡 Please note that for simplicity in this code, we will be using pykx. This is necessary as we need to import our ADF test function and plot a heatmap of our results.
+ 
+> `system "l pykx.q"`
+
+Next, we import the statsmodels.tsa.stattools library and define a custom cointegration function that returns a dictionary with the ADF test results given 2 assets. **We then apply our cointegration function to the Cartesian product of every pair of assets** to get a matrix with every p-value. 
+
+---
+
+![cointegration](https://github.com/hablapps/pairstrading/blob/5-Post/resources/CointAlGif.gif?raw=true)
+
+---
+
+```q
+// We import the cointegration function from statsmodels library in python
+coint:.pykx.import[`statsmodels.tsa.stattools]`:coint; 
+
+// Recieves 2 symbols and returns a dictionary
+fCoint: {p1:0f^(exec close from superTab where sym=x); // AssetX
+        p2: 0f^(exec close from superTab where sym=y); // AssetY
+        r: 0f^coint[p1;p2]`; // Coint results
+        `pair`score`pvalue`porcentages!(enlist (x,y);r[0];r[1];enlist r[2])}; // Get every result into a dictionary
+
+// We extract every distinct symbol
+symList: exec distinct sym from superTab;
+
+crossedList: symList cross symList
+
+matrix: fCoint .' crossedList;
+```
+
+### Code explanation
+
+1. The `coint` function **is imported from statsmodels**, thanks to the pykx library, which provides a cointegration tool in Python. This variable transforms every input data into its Python equivalent, facilitating the calculation of cointegration results.
+
+2. We then proceed to **create our custom cointegration function** called `fCoint`, which retrieves both asset prices from our table based on the input symbols. Subsequently, it utilizes the previously imported coint function to obtain the results and inserts them into a dictionary. (`symbol1symbol2...!(;v1;v2;...)`)
+
+3. We use QSQL to **extract every distinct symbol** from our table with the keyword [`exec`](https://code.kx.com/q/ref/exec/).
+
+> 💡[QSQL](https://code.kx.com/q/basics/qsql/) is a set of feautures that permits users to perform SQL-operations with a very similar syntax on tables in Q/kdb+. Which makes it pretty accesible for newcomers to this language.
+
+4. We utilize the `cross` operator, which conducts a [Cartesian product](https://en.wikipedia.org/wiki/Cartesian_product) on every symbol, **generating all possible pair combinations**.
+
+5. Finally, **we apply our function from step 2 to our list of every pair of symbols from step 4**. We can achive this by using the [apply operator](https://code.kx.com/q/ref/apply/) (`.`) follow by an [`each`](https://code.kx.com/q/ref/maps/#each) (`'`). So that we apply our function (`fCoint .' `) to each and every pair (`crossedList`)

```

Comment:
```suggestion
5. Finally, **we apply our function from step 2 to our list of every pair of symbols from step 4**. We can achieve this by using the [apply operator](https://code.kx.com/q/ref/apply/) (`.`) followed by an [`each`](https://code.kx.com/q/ref/maps/#each) (`'`). So that we apply our function (`fCoint .' `) to each pair (`crossedList`)

```